### DISCUSSION TITLE
How does implicit language activation work?

### DISCUSSION THREAD

**Discussion by winstliu** [microsoft/vscode-discussions#2442] [Extension Development QnA] [answered]
From https://code.visualstudio.com/api/references/activation-events, there is a note that says "Beginning with VS Code 1.74.0, languages contributed by your extension do not require a corresponding onLanguage activation event declaration for your extension to be activated".

I'm dealing with an extension that says it's being activated due to Rust:
![activated-by-onlanguage-rust](https://github.com/user-attachments/assets/1986e38c-41f4-4804-af97-a032ee6be5b6)

But it does _not_ contribute the Rust language.
![programming-languages-no-rust](https://github.com/user-attachments/assets/68420680-adf3-4c4e-89c2-6699b02d7d2b)

So what is causing the implicit activation to trigger on Rust?

**Comment by alefragnani** [comment]
Hi @winstliu ,

I'm not entirely sure, but since VS Code already supports Rust out of the box, maybe this extension only contributes `grammars` to the Rust language.

BTW, what does the `Activation Events` section says for this extension?

**Comment by winstliu** [comment, marked as answer]
Thanks for responding! I was able to figure this out - it was activated by an extension that depends on this one. _That_ extension has an `onLanguage:rust` activation event.

I found this behavior pretty confusing, so if I have the time I'll try to submit a VS Code issue about it.
